django/urls/base.py

```diff
@@ -179,3 +183,50 @@ def translate_url(url, lang_code):
             else:
                 url = urlunsplit((parsed.scheme, parsed.netloc, url, parsed.query, parsed.fragment))
     return url
+
+
+_PATH_PARAMETER_COMPONENT_RE = re.compile(
+    '<(?:(?P<converter>[^:]+):)?(?P<parameter>[A-Za-z0-9_]+)>'
```

Comment:
Agreed, when thinking about namespacing etc. this morning, I realized there was no fundamental reason for converter names to be identifiers.

It's probably a good sanity check, though. Enforcing that converter names are valid identifiers excludes special characters, preserving our ability to make backwards-compatible changes by giving a meaning to special characters such as `|` or `.`.